velox/connectors/hive/iceberg/tests/IcebergPartitionPathTest.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "velox/connectors/hive/iceberg/IcebergPartitionPath.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+namespace {
+
+template <typename T>
+std::string test(TransformType transform, T value, const TypePtr& type) {
+  return IcebergPartitionPath(transform).toPartitionString(value, type);
+}
+
+TEST(IcebergPartitionPathTest, identityInt32) {
+  EXPECT_EQ(test(TransformType::kIdentity, 100, INTEGER()), "100");
+  EXPECT_EQ(test(TransformType::kIdentity, -100, INTEGER()), "-100");
+  EXPECT_EQ(test(TransformType::kIdentity, 0, INTEGER()), "0");
+}
+
+TEST(IcebergPartitionPathTest, identityDate) {
+  EXPECT_EQ(test(TransformType::kIdentity, 18'262, DATE()), "2020-01-01");
+  EXPECT_EQ(test(TransformType::kIdentity, 0, DATE()), "1970-01-01");
+  EXPECT_EQ(test(TransformType::kIdentity, -1, DATE()), "1969-12-31");
+  EXPECT_EQ(test(TransformType::kIdentity, 2'932'897, DATE()), "10000-01-01");
+}
+
+TEST(IcebergPartitionPathTest, identityBoolean) {
+  EXPECT_EQ(test(TransformType::kIdentity, true, BOOLEAN()), "true");
+  EXPECT_EQ(test(TransformType::kIdentity, false, BOOLEAN()), "false");
+}
+
+TEST(IcebergPartitionPathTest, identityVarchar) {
+  EXPECT_EQ(
+      test(TransformType::kIdentity, StringView("a/b/c=d"), VARCHAR()),
+      "a/b/c=d");
+  EXPECT_EQ(test(TransformType::kIdentity, StringView(""), VARCHAR()), "");
+}
+
+TEST(IcebergPartitionPathTest, identityVarbinary) {
+  EXPECT_EQ(
+      test(
+          TransformType::kIdentity,
+          StringView("\x48\x65\x6c\x6c\x6f"),
```

Comment:
consider adding helper method testVarbinary that takes a single std::string argument. This will make code shorter and easier to read and allow testing long strings.